graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -48,79 +69,33 @@ def merge_schemas(schema_id_to_ast):
           same name as the type that it queries, if the schema contains type extensions or
           input object definitions, or if the schema contains mutations or subscriptions
         - SchemaNameConflictError if there are conflicts between the names of
-          types/interfaces/enums/scalars, or conflicts between the definition of directives
-          with the same name
+          types/interfaces/enums/scalars or fields, or conflicts between the definition of
+          directives with the same name
+        - InvalidCrossSchemaEdgeError if some cross schema edge provided lies within one schema,
```

Comment:
or if edges clash. For instance, let `cross_schema_edges` contain a 
```python
CrossSchemaEdgeDescriptor(
    edge_name = 'example_edge',
    outbound_side = FieldReference(
        schema_id = 'first',
        type_name = 'Human',
        field_name = 'id',
        ),
    inbound_side = FieldReference(
        schema_id = 'second',
        type_name = 'Person',
        field_name = 'identifier',
    ),
    out_edge_only=False,
)
```
Then `merge_schema` will attempt to create a new field for `Person` named `in_example_edge`. If Person already has a field `in_example_edge`, we have a clash.

Another way to generate a clash is if we do something like this but there are multiple types with the same edge name (e.g. type `City` might have an edge named `city_in` to both `Country` and `Region` types).

^^ It's up to you how much detail you want to include here, but I'd explicitly give an example of how a clash can happen since I was a bit confused by what clashing meant at first while reading through the tests.

Reply:
I'll add a little more to note the fact that field name conflicts can be due to fields created by cross schema edges.